### DISCUSSION TITLE
What's the recommended way to unit test Postgres queries, policies, when using Supabase

### DISCUSSION THREAD

**Discussion by Odas0R** [supabase/supabase#2003] [Questions] [answered]
First of all, I really like Supabase, a great product!

Most of the developers, including me, are not using fully their databases.

I've been creating DB tests on the back-end, and I wonder what are the best approaches, when using a service like Supabase?

Thanks.

**Comment by steve-chavez** [comment, marked as answer]
I'd recommend [pgTAP](https://pgtap.org/), It lets you write your tests in pure SQL.

There are some example tests [here](https://github.com/steve-chavez/socnet/tree/master/tests).

**Reply by Odas0R** [reply]
Ah cool, well that's way better.

Thanks!

**Reply by tawjaw** [reply]
How would you write unit test for RLS?
Can we use SET ROLE maybe?

**Reply by steve-chavez** [reply]
@tawjaw As done [here](https://github.com/steve-chavez/socnet/blob/master/tests/friendships.sql#L7-L13) with `SET LOCAL`.
